src/graphorm/index.ts

```diff
@@ -82,21 +81,24 @@ const checkIfTitleIsClickbait = (value?: string): boolean => {
 const createSmartTitleField = ({ field }: { field: string }): GraphORMField => {
   return {
     select: field,
-    transform: (value: string, ctx: Context, parent): string => {
+    transform: async (value: string, ctx: Context, parent) => {
       const typedParent = parent as {
-        ['i18nTitle']: I18nRecord;
-        ['smartTitle']: I18nRecord;
-        clickbaitShieldEnabled?: boolean;
+        i18nTitle: I18nRecord;
+        smartTitle: I18nRecord;
         clickbaitProbability?: string;
         [key: string]: unknown;
       };
 
+      const settings = await ctx.dataLoader.userSettings.load({
+        userId: ctx.userId!,
+      });
```

Comment:
It would be awesome if we could also reuse this in graphorm, but I guess that would require fiddling more on graphorm side to replace queries there with data loader result. But I think this should be better overall.

Reply:
What would be ideal case in your opinion? graphorm is a translation layer from graphql to pgsql.
not sure how dataloader fits to this picture

Reply:
if we could make hints for reusable entities so that graphorm can maybe fetch those through data loader instead of nested sub query

Another example for source:
- we fetch feed for source X
- in that response we nested query source for each post (it will be the same source X)
- we return the response

With data loader:
- we fetch feed for source X
- we hint to the graphorm that source should be fetched through data loader
- we execute the query in parallel with dataloader requesting source
- we merge and reuse data loader response in place of all source sub queries

It could be a nice caching layer inside graphorm when we have a lot of nested fields and complexity of resolving those rises.

Something like that was on my mind.